### PR TITLE
setup.py: move more modules back to "install_requires"

### PR DESCRIPTION
**Pull request by tchaikov**
Signed-off-by: Kefu Chai <kchai@redhat.com>

### CODE REVIEW

**Comment by kshtsk** [comment]
@susebot run deploy

**Comment by tchaikov** [comment]
@kshtsk could you please rerun deploy test? i just updated the PR.

File: teuthology/provision/__init__.py [hunk 1]
```diff
@@ -20,6 +16,7 @@ def _logfile(ctx, shortname):
 
 
 def reimage(ctx, machine_name):
+    from teuthology.provision import fog
```
Comment (kshtsk) [inline teuthology/provision/__init__.py]: Why is this required, that is not a good practice, looks like you're trying to resolve circular imports here, where are you facing this issue.

Comment (tchaikov) [reply teuthology/provision/__init__.py]: no. i am not trying to fix cyclic dependency issue. please see the commit message. i am trying to minimize the set of dependencies required for running `vstart_runner.py`, which does not need to talk to a remote service for provision, etc.

Comment (kshtsk) [reply teuthology/provision/__init__.py]: I don't see how it could talk to remote service for provision until you call corresponding methods, what's wrong with those dependencies in the vstart? Still I don't think to place imports in the function is a good idea.

Comment (tchaikov) [reply teuthology/provision/__init__.py]: no, it does not use the openstack API until the methods are called, but the crux is that i want to avoid the import. because it's a waste of time and resource to install unused modules, just because they are imported in the beginning of the `__init__.py`.

> Still I don't think to place imports in the function is a good idea.

could you be more specific why it is not a good idea? yes, it's unusual. but why is it bad? if you insist, i will have to partially invert my changes in `setup.py` and move some "orchestra" back into "install_requires".

**Comment by tchaikov** [comment]
jenkins retest please

File: teuthology/provision/__init__.py (hunk 1 above)
Comment (tchaikov) [reply teuthology/provision/__init__.py]: seems some unit tests are using them. so i end up moving them to "install_requires".

**Comment by tchaikov** [comment]
@susebot run deploy

**Comment by kshtsk** [comment]
@tchaikov I’m sorry not keeping the discussion since I’m on vacation till the second week of January, so I cannot work on this PR. But I insist on keep import on the beginning of the module, having imports in functions is very special case and it should be very good reason (improving dependency deployment speed for vstart is not the one).

**Comment by kshtsk** [comment]
Btw, do we have a tracker ticket for this?

**Comment by tchaikov** [comment]
@kshtsk no worries. i dropped that change. hopefully, this change is less controversial now. AFAIK, no, we don't have a tracker ticket for it.

**Comment by callithea** [comment]
> @kshtsk no worries. i dropped that change. hopefully, this change is less controversial now. AFAIK, no, we don't have a tracker ticket for it.

@tchaikov @kshtsk https://tracker.ceph.com/issues/43331

**Comment by kshtsk** [comment]
thanks guys, feel free to merge this... looks like the deploy test has got a limited functionality after the default os version changed to centos8 and we do not have centos8 image in our suse tenant of openstack. I can work on fixing it after I get back from vacation.

**Comment by tchaikov** [comment]
thanks @kshtsh. and enjoy your vacation!
